TicTacToeApp/Board.swift

```diff
@@ -0,0 +1,48 @@
+
+public class Board {
+  
+  private let markedSpaces: [Int: PlayerMark]
+  private let size = 3
+
+  public init() {
+    markedSpaces = [:]
+  }
+
+  public init(markedSpaces: [Int : PlayerMark]) {
+    self.markedSpaces = markedSpaces
+  }
+
+  public func markSpace(id: Int, mark: PlayerMark) -> Board {
+    var newSpaces = markedSpaces
+    newSpaces[id] = mark
+
+    return Board(markedSpaces: newSpaces)
+  }
+
+  public func readSpace(id: Int) -> PlayerMark? {
+    return markedSpaces[id]
+  }
+
+  public func allSpacesMarked() -> Bool {
```

Comment:
allMarked(), full(), or filled() come to mind. I'll rename to allMarked() unless you feel strongly about an alternative.